syntax = "proto3";

package geodesic_gRPC;

// The greeting service definition.
service Geodesic {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  // request shortest path using vertex id
  rpc FindPathByVertexCord(FindPathByVertexCordRequest) returns (Path) {}

  rpc GetTerrainInfo(HelloRequest) returns(TerrainInfo) {}

  rpc SimplifyTerrain(SimplifyTerrainRequest) returns(ModelPath){}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

// request shortest path using vertex id
message FindPathByVertexCordRequest {
    string algo_type = 1;
    string v1 = 2;
    string v2 = 3;
}

message SimplifyTerrainRequest {
  float beta = 1;
  string old_model_path = 2;
}

// the response containing the path
message Path {
    repeated float path = 1;
    string message = 2;
}

message TerrainInfo{
  string info = 1;
}

message ModelPath {
  string model_path = 1;
  
}